viabel/optimizing_classes.py

```diff
@@ -0,0 +1,237 @@
+from abc import ABC, abstractmethod
+import tqdm
+import autograd.numpy as np
+from scipy.stats import t as tdist
+from viabel.approximations import MFGaussian
+
+
+
+class Optimizer(ABC):
+    """An abstract class for optimization
+    """
+    
+    @abstractmethod
+    def optimize(self, n_iters, objective, init_param, smoothed_prop=0.2):
```

Comment:
smoothed_prop shouldn't be an argument to the abstract method. Have **kwargs instead